File: device/src/main/java/com/blackstone/device/notification/NotificationsImpl.java
package com.blackstone.device.notification;

import android.app.Notification;
import androidx.annotation.NonNull;
import androidx.core.app.NotificationManagerCompat;

public final class NotificationsImpl implements Notifications {

    private final NotificationManagerCompat notificationManagerCompat;

    public NotificationsImpl(@NonNull final NotificationManagerCompat notificationManagerCompat) {
        this.notificationManagerCompat = notificationManagerCompat;
    }

    @Override
    public void showNotification(final int notificationId, final Notification notification) {
        notificationManagerCompat.notify(notificationId, notification);
    }

    @Override
    public void updateNotification(final int notificationId, final Notification notification) {
        notificationManagerCompat.notify(notificationId, notification);
    }

    @Override
    public void hideNotification(final int notificationId) {
        notificationManagerCompat.cancel(notificationId);
    }

    @Override
    public void hideNotifications() {
        notificationManagerCompat.cancelAll();
    }
}

File: device/src/main/java/com/blackstone/device/notification/NotificationsImpl.kt
package com.blackstone.device.notification

import androidx.core.app.NotificationManagerCompat
import com.blackstone.device.notification.Const.NOTIFICATION_ID_FORPOST_CLOSE
import com.blackstone.device.notification.Const.NOTIFICATION_ID_FORPOST_TP
import com.blackstone.device.notification.Const.NOTIFICATION_ID_OCTAL_CLOSE
import com.blackstone.device.notification.Const.NOTIFICATION_ID_OCTAL_TP
import com.blackstone.domain.entity.Config
import com.blackstone.domain.repositories.AppRepository

class NotificationsImpl (
    private val notificationManagerCompat: NotificationManagerCompat,
    private val notificationBuilder: NotificationBuilder,
    repository: AppRepository
): Notifications {

    private var configApp: Config = repository.getSharedProviderService().getNotificationSettings()

    override fun showNotification(notificationId: Int) {
        when (notificationId) {
            NOTIFICATION_ID_FORPOST_CLOSE -> if (configApp.closeForpost) notificationManager(notificationId)
            NOTIFICATION_ID_OCTAL_CLOSE -> if (configApp.closeOctal) notificationManager(notificationId)
            NOTIFICATION_ID_FORPOST_TP -> if (configApp.tpForpost) notificationManager(notificationId)
            NOTIFICATION_ID_OCTAL_TP -> if (configApp.tpOctal) notificationManager(notificationId)
        }
    }

    private fun notificationManager(notificationId: Int) {
        notificationManagerCompat.notify(notificationId,notificationBuilder.builder(notificationId))
    }
}